src/pages/task/metadata/Stepback.tsx

```diff
@@ -0,0 +1,75 @@
+import styled from "@emotion/styled";
+import Badge from "@leafygreen-ui/badge";
+import Tooltip from "@leafygreen-ui/tooltip";
+import { Body } from "@leafygreen-ui/typography";
+import Icon from "components/Icon";
+import { MetadataItem } from "components/MetadataCard";
+import { size } from "constants/tokens";
+import { TaskQuery } from "gql/generated/types";
+import { useBreakingTask } from "hooks/useBreakingTask";
+import { useLastPassingTask } from "hooks/useLastPassingTask";
+
+type Props = {
+  taskId: string;
+};
+
+/**
+ * Returns whether the task is in stepback.
+ * @param task The task to check if it is in stepback.
+ * @returns Whether the task is in stepback.
+ */
+export function inStepback(task: TaskQuery["task"]) {
+  // lastFailingStepbackTaskId is set only in the middle/last of stepback (not the first task).
+  const stepback =
+    task?.stepbackInfo?.lastFailingStepbackTaskId !== undefined &&
+    task?.stepbackInfo?.lastFailingStepbackTaskId !== "";
+
+  // nextStepbackTaskId is set when the next task in stepback in known, in the beginning
+  // of stepback, it is known right away. In the rest of stepback, it is not.
```

Comment:
nit. there might be a few typos in this comment e.g. `in stepback in known`. `in the beginning of...` seems like it should be its own sentence as well